tests/simulator_test.cpp

```diff
@@ -1,2 +1,50 @@
+#include "constants.h"
+#include "main/simulator.h"
 #include "gtest/gtest.h"
 
+using constants::kDefaultCollectorPopulation;
+using constants::kDefaultDepth;
+using constants::kDefaultEnergyCollectorRadius;
+using constants::kDefaultNumRaysPerRow;
+using constants::kDefaultSimulationRadius;
+using constants::kDefaultSourcePower;
+using constants::kSkipFreq;
+using constants::kSkipPath;
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+
+const Vec3 kDefaultSourcePosition(0, 0, constants::kDefaultSimulationRadius);
+const Vec3 kVecStraightDown(0, 0, -1);
+const float kDefaultEnergyPerRay =
+    constants::kDefaultSourcePower /
+    (constants::kDefaultNumRaysPerRow * constants::kDefaultNumRaysPerRow);
+
+class ReferencePlateTest : public ::testing::Test {
+public:
+  ReferencePlateTest()
+      : sim(Simulator(kSkipFreq, kDefaultNumRaysPerRow,
```

Comment:
Do you realize that you create the temp. object by calling Simulator(...) and then calling copy ctor. by calling: `sim_(Simulator(...))` right? Just drop the `Simulator()` and it will work